immich-app/immich
# feat(web,server)!: configure machine learning via the UI

# Breaking Change
The way we disabled machine learning has changed in this pull request.

To disable machine learning now, use `IMMICH_MACHINE_LEARNING_ENABLED=false` 

(previously `IMMICH_MACHINE_LEARNING_URL=false`)
<hr>

In this PR:

- Make machine learning url configurable via the admin settings (still can be set/disabled via `IMMICH_MACHINE_LEARNING_URL`)
- Add options to disable certain ML features in the UI (these skip corresponding jobs)
- Remove unused search config endpoint (replaced with the new features endpoint)
- Update jobs to become disabled with corresponding feature flags

Tested scenarios:
- Start up with ML disabled via env verify jobs are disabled and ML doesn't run on upload
- Start up with ML enabled and disabled via the UI and verify ML stops running
- Start up with ML enabled and disable individual features. Verify associated jobs stop running
- Start up with ML enabled and verify all jobs still function correctly

![image](https://github.com/immich-app/immich/assets/4334196/214c7733-308b-4b52-8e5d-58059b01b929)
